webapp/components/notifications/NotificationMenu/NotificationMenu.vue

```diff
@@ -46,24 +48,53 @@ export default {
           variables,
         })
         if (!(markAsRead && markAsRead.read === true)) return
-        this.notifications = this.notifications.map(n => {
-          return n.from.id === markAsRead.from.id ? markAsRead : n
+        this.displayedNotifications = this.displayedNotifications.map(n => {
+          return this.equalNotification(n, markAsRead) ? markAsRead : n
         })
       } catch (err) {
-        throw new Error(err)
+        this.$toast.error(err.message)
       }
     },
+    equalNotification(a, b) {
+      return a.from.id === b.from.id && a.createdAt === b.createdAt && a.reason === b.reason
+    },
   },
   computed: {
-    totalNotifications() {
-      return (this.notifications || []).length
+    notificationsCount() {
+      return (this.displayedNotifications || []).length
+    },
+    unreadNotificationsCount() {
+      let countUnread = 0
+      if (this.displayedNotifications) {
+        this.displayedNotifications.forEach(notification => {
+          if (!notification.read) countUnread++
+        })
+      }
+      return countUnread
     },
```

Comment:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-23T11:08:07Z" title="Monday, September 23rd 2019, 1:08:07 pm +02:00">Sep 23, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -35,7 +49,39 @@ export default {
   props: {
     placement: { type: String },
   },
+  created() {
+    setInterval(this.updateNotifications, REFRESH_MILLISEC)
+  },
+  destroyed() {
+    clearInterval(this.updateNotifications)
+  },
   methods: {
+    async updateNotifications() {
+      try {
+        const {
+          data: { notifications },
+        } = await this.$apollo.mutate({
+          mutation: notificationQuery(this.$i18n),
+        })
+        // add all the new notifications to the old notifications at top of the list
+        if (notifications) {
+          notifications.forEach(udatedListNotification => {
+            const sameNotification = this.notifications.find(function(oldListNotification) {
+              return (
+                oldListNotification.from.id === udatedListNotification.from.id &&
+                oldListNotification.createdAt === udatedListNotification.createdAt &&
+                oldListNotification.reason === udatedListNotification.reason
+              )
+            })
+            if (sameNotification === undefined) {
+              this.notifications.unshift(udatedListNotification)
+            }
+          })
+        }
```


I find this code way to complex. Can't you just append two arrays and the remove duplicates based on a criteria? I googled and found https://medium.com/dailyjs/how-to-remove-array-duplicates-in-es6-5daa8789641c there is also one by Mozilla using `reduce` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce#Remove_duplicate_items_in_array

Reply:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-09-23T13:59:48Z" title="Monday, September 23rd 2019, 3:59:48 pm +02:00">Sep 23, 2019</time>_

---


All your examples @roschaefer are no really working, because if the `read` property changes in my code it should be equal as well. And therefore the new fetched notification should not be added to the beginning …